### PR TITLE
fix: iterator cursor progress too fast(#36179)

### PR DESCRIPTION
**Pull request by MrPresent-Han**
related: #36179

### CODE REVIEW

File: internal/proxy/task_query.go [hunk 1]
```diff
@@ -343,7 +364,7 @@ func (t *queryTask) PreExecute(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
-	t.RetrieveRequest.ReduceStopForBest = queryParams.reduceStopForBest
+	t.RetrieveRequest.ReduceType = int32(queryParams.reduceType)
```
Comment (congqixia) [inline internal/proxy/task_query.go:367]: Maybe we shall fill both field value for backward compatibility concern?

Comment (MrPresent-Han) [reply internal/proxy/task_query.go:367]: compatibility in the process of rolling upgrade?
this may only result in the number of results not than 'many'

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**